src/geo/transform.js

```diff
@@ -862,8 +863,33 @@ class Transform {
                 const minLat = latFromMercatorY((it.y + 1) / tilesAtZoom);
                 const maxLat = latFromMercatorY((it.y) / tilesAtZoom);
                 const closestLat = Math.min(Math.max(centerLatitude, minLat), maxLat);
-                const scale = circumferenceAtLatitude(closestLat) / circumferenceAtLatitude(centerLatitude);
-                tileScaleAdjustment = Math.min(scale, 1.0);
+
+                let relativeTileScale = circumferenceAtLatitude(closestLat) / circumferenceAtLatitude(centerLatitude);
+
+                // With globe, the rendered scale does not exactly match the mercator scale at low zoom levels.
+                // Account for this difference during LOD of loading so that you load the correct size tiles.
+                // We try to compromise between two conflicting requirements:
+                // - loading tiles at the camera's zoom level (for visual and styling consistency)
+                // - loading correct size tiles (to reduce the number of tiles loaded)
+                // These are arbitrarily balanced:
+                if (closestLat === centerLatitude) {
+                    // For tiles that are in the middle of the viewport, prioritize matching the camera
+                    // zoom and allow divergence from the true scale.
+                    const maxDivergence = 0.3;
+                    tileScaleAdjustment = 1 / Math.max(1, this._mercatorScaleDifference - maxDivergence);
+                } else {
+                    // For other tiles, use the real scale to reduce tile counts near poles.
+                    tileScaleAdjustment = Math.min(1, relativeTileScale / this._mercatorScaleDifference);
+                }
+
+                // Ensure that all tiles near the center have the same zoom level.
+                // With LOD tile loading, tile zoom levels can change when scale slightly changes.
+                // These differences can be pretty different in globe view. Work around this by
+                // making more tiles match the center tile's zoom level. If the tiles are nearly big enough,
+                // round up.
+                if (this.zoom <= 5 && it.zoom === maxZoom - 1 && relativeTileScale >= 0.9) {
```

Comment:
For `this.zoom <= 5`, is this somewhat chosen based on the zoom transition threshold or the range [3, 5] [defined here](https://github.com/mapbox/mapbox-gl-js/pull/11951/files#diff-42ed1ce9d0fab96619fdbf32aa6890a317cc995f7be075bd977fffb0a4cf9ac3R43), or is it independent? e.g. would changing the zoom transition threshold or range impact this negatively?